internal/dag/builder.go

```diff
@@ -917,29 +919,89 @@ func (b *Builder) processTCPProxy(sw *ObjectStatusWriter, ir *ingressroutev1.Ing
 
 		// follow the link and process the target ingress route
 		sw, commit := sw.WithObject(dest)
-		b.processTCPProxy(sw, dest, visited, host)
+		b.processIngressRouteTCPProxy(sw, dest, visited, host)
 		commit()
 	}
 }
 
-func (b *Builder) processTCPProxyHTTPProxy(sw *ObjectStatusWriter, httpproxy *projcontour.HTTPProxy, host string) {
-	if len(httpproxy.Spec.TCPProxy.Services) > 0 {
+func (b *Builder) processHTTPProxyTCPProxy(sw *ObjectStatusWriter, httpproxy *projcontour.HTTPProxy, visited []*projcontour.HTTPProxy, host string) bool {
```

Comment:
Maybe add a comment to the method describing what `bool` means? That's the only confusing part I noticed when reading through. Otherwise looks good. =)

Reply:
done